main.c

```diff
@@ -56,18 +76,76 @@ int main(int argc, char** argv) {
     // initialize timer
     timer0_init();
     
+    /***********set up CAN**********/
+
+    // Set up CAN TX
+    TRISC0 = 0;
+    RC0PPS = 0x33;
+
+    // Set up CAN RX
+    TRISC1 = 1;
+    ANSELC1 = 0;
+    CANRXPPS = 0x11;
+
+    // set up CAN module
+    can_timing_t can_setup;
+    can_generate_timing_params(_XTAL_FREQ, &can_setup);
+
+    can_init(&can_setup, can_msg_handler);
+
+    // set up CAN tx buffer
+    txb_init(tx_pool, sizeof(tx_pool), can_send, can_send_rdy);
+    
     // initialize misc variables
     fill_level = 0;
+    direction = 0;
     uint32_t last_millis = millis();
 
-    // keep the board running and listening for interrupts
+    // main loop
     while (true) {
-        // heartbeat LED
-        if (millis() - 1000 > last_millis){
+        
+        if(millis() > last_millis + MAX_LOOP_TIME_DIFF_ms) {
             last_millis = millis();
+            
+            // heartbeat LED
             BLUE = !BLUE;
+            
+            can_msg_t fill_msg;
+            build_fill_msg(millis(),
+                           fill_level,
+                           direction,
+                           &fill_msg);
+            
+            txb_enqueue(&fill_msg);
```

Comment:
There is a soft rule that all boards should send a GENERAL_BOARD_STATUS message every second, with E_NOMINAL if everything is going well. Radio board uses this to report how many boards are currently active in the rocket, and whether any of them are showing errors (or at least, it's supposed to report this, I'm not sure if we ever got that working).

Here's radio's source file for checking whether boards are present and OK:  https://github.com/waterloo-rocketry/cansw_radio/blob/32596bdc1e8dafad1fca0e2b322a53e3afc48afd/sotscon.c#L18

see vent doing this: https://github.com/waterloo-rocketry/cansw_vent/blob/020f12489f8bcc6523c28edb3b0a076923bd8aba/main.c#L219